internal/services/storage/storage_share_resource.go

```diff
@@ -384,39 +576,75 @@ func resourceStorageShareUpdate(d *pluginsdk.ResourceData, meta interface{}) err
 func resourceStorageShareDelete(d *pluginsdk.ResourceData, meta interface{}) error {
 	storageClient := meta.(*clients.Client).Storage
 	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	fileSharesClient := meta.(*clients.Client).Storage.ResourceManager.FileShares
 	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := shares.ParseShareID(d.Id(), storageClient.StorageDomainSuffix)
-	if err != nil {
-		return err
+	if !features.FivePointOhBeta() {
+		if strings.HasPrefix(d.Id(), "/subscriptions/") {
```

Comment:
Same comment here about squishing the conditions together like we do for the Create/Update/Read